infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/TrackMeterHeights.kt

```diff
@@ -0,0 +1,281 @@
+    return allTicks.filterIndexed { i, (trackMeterInKm, segmentIndex) ->
+        tickIsGoodToSend(segmentIndex, i, trackMeterInKm, allTicks, minTickSpace)
+    }
+}
+
+// Logic for visualization: prevent height ticks from getting too close to each other. Ordinary
+// ticks always have room between each other, we just want to prevent them from getting too close to
+// segment boundary ticks for clarity.
+private fun tickIsGoodToSend(
+    segmentIndex: Int?,
+    i: Int,
+    trackMeterInKm: BigDecimal,
+    allTicks: List<TrackMeterHeightTick>,
+    minTickSpace: BigDecimal,
+): Boolean {
+    val isSegmentBoundaryPoint = segmentIndex != null
+    val isKmStartTick = i == 0
+    val hasSpaceAfterPreviousTick = i > 0 && trackMeterInKm - allTicks[i - 1].trackMeterInKm >= minTickSpace
+    val hasSpaceBeforeNextTick =
+        (i == allTicks.lastIndex || allTicks[i + 1].trackMeterInKm - trackMeterInKm >= minTickSpace)
+
+    return isSegmentBoundaryPoint || isKmStartTick || (hasSpaceAfterPreviousTick && hasSpaceBeforeNextTick)
+}
+
+private fun fitKmTicksWithinAlignmentBounds(
+    kmNumber: KmNumber,
+    alignmentStart: AddressPoint,
+    alignmentEnd: AddressPoint,
+    ticks: List<TrackMeterHeightTick>,
+): List<TrackMeterHeightTick> {
+    // Special-case first and last alignment points so we get as close as possible to the track ends
+    val alignmentStartTick =
+        if (kmNumber == alignmentStart.address.kmNumber) TrackMeterHeightTick(alignmentStart.address.meters, null)
+        else null
+    val alignmentEndTick =
+        if (kmNumber == alignmentEnd.address.kmNumber) TrackMeterHeightTick(alignmentEnd.address.meters, null) else null
+
+    val ticksStrictlyAfterStart =
+        if (alignmentStartTick != null)
+            ticks.dropWhile { tick -> tick.trackMeterInKm <= alignmentStartTick.trackMeterInKm }
+        else ticks
+
+    val ticksStrictlyWithinAlignment =
+        if (alignmentEndTick != null)
+            ticksStrictlyAfterStart.takeWhile { tick -> tick.trackMeterInKm < alignmentEndTick.trackMeterInKm }
+        else ticksStrictlyAfterStart
+
+    return listOfNotNull(alignmentStartTick) + ticksStrictlyWithinAlignment + listOfNotNull(alignmentEndTick)
+}
+
+private fun getTickTrackLocationsByKm(
+    geocodingContext: GeocodingContext,
+    alignment: IAlignment,
+    ticksByKm: List<List<TrackMeterHeightTick>>,
+    referencePointIndices: IntRange,
+): List<List<AddressPoint>> {
+    val trackAddresses =
+        ticksByKm.zip(referencePointIndices) { kmTicks, referencePointIndex ->
+            val km = geocodingContext.referencePoints[referencePointIndex].kmNumber
+            kmTicks.map { tick -> TrackMeter(km, tick.trackMeterInKm) }
+        }
+    return processFlattened(trackAddresses) { allTrackAddresses ->
+        // null-safety: fitKmTicksWithinAlignmentBounds drops all ticks outside the alignment
+        // bounds, so everything is geocodable.
```

Comment:
Onko tämä kommentti unohtunut päivittää tai koskeeko kommentti jotain funktiota jota kutsutaan jossain syvemmällä? En oikein itse löydä nopealla lukemisella yhteyttä tämän funktion ja `fitKmTicksWithinAlignmentBounds()`:in välillä